webrtc/src/rtp_transceiver/rtp_sender/mod.rs

```diff
@@ -75,19 +72,26 @@ impl RTPSenderInternal {
     }
 }
 
-/// RTPSender allows an application to control how a given Track is encoded and transmitted to a remote peer
-pub struct RTCRtpSender {
+pub struct TrackEncoding {
     pub(crate) track: Mutex<Option<Arc<dyn TrackLocal + Send + Sync>>>,
 
     pub(crate) srtp_stream: Arc<SrtpWriterFuture>,
+
+    pub(crate) rtcp_reader: Arc<dyn RTCPReader + Send + Sync>,
     pub(crate) stream_info: Mutex<StreamInfo>,
 
     pub(crate) context: Mutex<TrackLocalContext>,
 
+    pub(crate) ssrc: SSRC,
+}
+
+/// RTPSender allows an application to control how a given Track is encoded and transmitted to a remote peer
+pub struct RTCRtpSender {
+    pub(crate) track_encodings: RwLock<Vec<Arc<TrackEncoding>>>,
```

Comment:
I don't expect you need to use an async lock here. As long as the await point isn't being held across `await`s a regular sync lock is preferred. See https://tokio.rs/tokio/tutorial/shared-state#on-using-stdsyncmutex

Reply:
Hadn't even occured to me which RwLock I'd imported, I'll watch for that in the future

Reply:
Yeah generally there are a lot of asynchronous locks in this project that should be reconsidered. I suspect a big chunk of these can become sync